### PR TITLE
[TT-8298] Fixing bug of empty APIName and APIID in access_rights

### PR DESCRIPTION
**Pull request by jeffy-mathew**
The tag of APIName and APIID was wrong because of which API was returning empty string

## Description

https://tyktech.atlassian.net/browse/TT-8298
https://github.com/TykTechnologies/tyk/issues/4350

in house PR for https://github.com/TykTechnologies/tyk/pull/4854  (to run CI)

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)

## Checklist

- [ ] I ensured that the documentation is up to date
- [ ] I explained why this PR updates go.mod in detail with reasoning why it's required
- [ ] I would like a code coverage CI quality gate exception and have explained why

### CODE REVIEW

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/5260/merge
Commit:  
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5444627974)

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/5260/merge
Commit:  
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5444627974)

File: gateway/policy.go [hunk 1]
```diff
@@ -18,8 +18,8 @@ import (
 )
 
 type DBAccessDefinition struct {
-	APIName              string                       `json:"apiname"`
-	APIID                string                       `json:"apiid"`
+	APIName              string                       `json:"api_name"`
```
Comment (furkansenharputlu) [inline gateway/policy.go:21]: I think this lacks bson too

Comment (jeffy-mathew) [reply gateway/policy.go:21]: it's not been used in dashboard. 
dashboard maintains separate structs for this and that's root cause for this disconnect.

**Review by furkansenharputlu — APPROVED** [review]

**Pull request merged**